src/main/java/org/sagebionetworks/web/client/security/AuthenticationControllerImpl.java

```diff
@@ -423,4 +420,27 @@ public void onSuccess(UserProfile result) {
       }
     );
   }
+
+  private void setPersistentLocalStorageKeys() {
+    String[] swcPersistentLocalStorageKeys = new String[] {
+      USER_AUTHENTICATION_RECEIPT,
+      NIH_NOTIFICATION_DISMISSED,
+      COOKIES_ACCEPTED,
+    };
+    String[] srcPersistentLocalStorageKeys = jsniUtils.getSrcPersistentLocalStorageKeys();
+
+    int srcKeysLength = srcPersistentLocalStorageKeys.length;
+    int swcKeysLength = swcPersistentLocalStorageKeys.length;
+
+    this.persistentLocalStorageKeys = new String[srcKeysLength + swcKeysLength];
+
+    for (int i = 0; i < srcKeysLength; i++) {
+      this.persistentLocalStorageKeys[i] = srcPersistentLocalStorageKeys[i];
+    }
+
+    for (int i = 0; i < swcKeysLength; i++) {
+      this.persistentLocalStorageKeys[srcKeysLength + i] =
+        swcPersistentLocalStorageKeys[i];
```

Comment:
If we used a List<String>, I wouldn't have to think about if `srcKeysLength + i` is the right index